### PR TITLE
Adapt Elemental test for BCI release pipeline

### PR DESCRIPTION
**Pull request by ldevulder**
Verification runs were done on my local lab.

Here some screenshots for validation. Anyway these are new tests which only impact Elemental images.

Job status:
Screenshot from 2024-07-25 15-45-13

Image generation:
Screenshot from 2024-07-25 15-46-01

Image validation:
Screenshot from 2024-07-25 15-44-36

### CODE REVIEW

File: tests/elemental/container_validation.pm [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+        assert_script_run("qemu-img convert -f raw -O qcow2 $shared/$flavor-$arch.raw $shared/elemental-$flavor-$arch.qcow2");
+        assert_script_run("qemu-img resize $shared/elemental-$flavor-$arch.qcow2 20G");
+        upload_asset("$shared/elemental-$flavor-$arch.qcow2", 1);
+    }
+
+    if ($image =~ m/iso-image/) {
+        # Create and upload ISO image
+        record_info('ISO', 'Generate and upload ISO');
+        assert_script_run("podman run --rm -v $shared:/host:Z $image /bin/sh -c 'busybox cp /elemental-iso/*.iso /host/elemental-$flavor-$arch.iso'");
+        upload_asset("$shared/elemental-$flavor-$arch.iso", 1);
```
Comment (jlausuch) [inline tests/elemental/container_validation.pm:102]: I am not sure where this asset will be uploaded, in the worker or in OSD.. 
@okurz can you confirm please?
(Context: the idea of this module is to upload an asset to be used by a chained job)

Comment (okurz) [reply tests/elemental/container_validation.pm:102]: This uploads the file to the openQA webUI over the worker. Reference: http://open.qa/api/testapi/#_upload_asset

File: tests/elemental/container_validation.pm [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+        }
+
+        record_info('Grub', 'Test that /etc/default/grub exists and it is not empty');
+        assert_script_run("podman exec $cnt_name /bin/sh -c 'test -s /etc/default/grub'");
+
+        record_info('Repos', 'Image should come with empty repos');
+        validate_script_output("podman exec $cnt_name /bin/sh -c 'zypper lr' 2>&1", sub { m/No repositories defined/ }, proceed_on_failure => 1);
+        assert_script_run("podman exec $cnt_name /bin/sh -c '[[ -z \"\$(ls -A /etc/zypp/repos.d)\" ]]'");
+
+        unless ($image =~ m/(base|kvm)-os-container/) {
```
Comment (kkaempf) [inline tests/elemental/container_validation.pm:65]: This test should be improved and not rely on specific container names.

(not sure if this test is needed at all - missing firmware would mean zypper or rpm are defective)

File: tests/elemental/container_validation.pm [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+    if ($image =~ m/os-container/) {
+        assert_script_run("podman pull $image");
+        assert_script_run("podman run --name $cnt_name -v $shared:/host:Z -dt $image sleep infinity");
+
+        record_info('Kernel', 'Test that kernel files are present');
+        foreach my $file ('initrd', 'vmlinuz') {
+            validate_script_output("podman exec $cnt_name /bin/sh -c 'ls /boot'", sub { m/$file/ });
+        }
+
+        record_info('Grub', 'Test that /etc/default/grub exists and it is not empty');
```
Comment (kkaempf) [inline tests/elemental/container_validation.pm:58]: if /etc/default/grub doesn't exist would mean the grub package is missing.

openqa shouldn't test existance of packages (resp. if packages are properly installed) imho.

**Review by kkaempf — APPROVED** [review]
will probably work but tests are too extensive imho

File: tests/elemental/container_validation.pm (hunk 3 above)
Comment (ldevulder) [reply tests/elemental/container_validation.pm:58]: This was an already existing test and I think that it has been added after the issue we had: shipping some Elemental images with a Grub issue. But can be removed later for sure. But I don't have time to do this now (vacations ;-)).

File: tests/elemental/container_validation.pm (hunk 2 above)
Comment (ldevulder) [reply tests/elemental/container_validation.pm:65]: Same as previous comment: this was an already existing test. And yes can be removed later if it's not needed.

As for the the image check I keep the same code as before, but yes I think we can use what's in `/etc/os-release` to identify the correct image.

**Review by jlausuch — APPROVED** [review]

**Pull request merged**